nemo_rl/algorithms/dpo.py

```
@@ -644,15 +644,31 @@ def dpo_train(
+                        ) or full_metric_name.startswith("val:"), (
+                            f"metric_name={full_metric_name} must start with 'val:' or 'train:'"
+                        )
+                        parts = full_metric_name.split(":")
+                        train_or_val = "val" if "val" in parts[0] else "train"
+                        metric_name = parts[1]
+
+                        metrics_source = (
+                            metrics if train_or_val == "train" else val_metrics
+                        )
+                        if not metrics_source or metric_name not in metrics_source:
                             warnings.warn(
```

Ruff (0.14.0)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)